utils/src/main/java/software/amazon/awssdk/utils/async/SimplePublisher.java

```diff
@@ -185,9 +171,7 @@ public CompletableFuture<Void> error(Throwable error) {
         OnErrorQueueEntry<T> entry = new OnErrorQueueEntry<>(error);
 
         try {
-            validateRejectState();
-            setRejectExceptionOrThrow(() -> new IllegalStateException("error() has been invoked"));
-            eventQueue.add(entry);
+            standardPriorityQueue.add(entry);
```

Comment:
Should it be `highPriorityQueue.add(entry)`?

Reply:
Hmm... it could be, but that's not what the design is. The javadoc implies it should be standardPriorityQueue as well. It's the difference of whether we want to preserve the relative order of send() and error() calls. We could change the semantics of the API to have error() calls skip previous send() calls, but then the caller wouldn't know the order of messages the downstream subscriber receives when errors are involved.

If we really wanted a "no really, die immediately", we could add a high-priority `error()` method. I don't know if we'd use it.

Reply:
I see, looks like in the existing implementation, it does not skip other data events if there's an error event. I think I missed/forgot that part. Can we add some documentation for `highPriorityQueue`?